apps/jsonrpc2/lib/jsonrpc2/server/handler.ex

```diff
@@ -0,0 +1,283 @@
+defmodule JSONRPC2.Server.Handler do
+  @moduledoc """
+  A transport-agnostic server handler for JSON-RPC 2.0.
+
+  ## Example
+
+      defmodule SpecHandler do
+        use JSONRPC2.Server.Handler
+
+        def handle_request("subtract", [x, y]) do
+          x - y
+        end
+
+        def handle_request("subtract", %{"minuend" => x, "subtrahend" => y}) do
+          x - y
+        end
+
+        def handle_request("update", _) do
+          :ok
+        end
+
+        def handle_request("sum", numbers) do
+          Enum.sum(numbers)
+        end
+
+        def handle_request("get_data", []) do
+          ["hello", 5]
+        end
+      end
+
+      SpecHandler.handle(~s({"jsonrpc": "2.0", "method": "subtract", "params": [42, 23], "id": 1}))
+      #=> ~s({"jsonrpc": "2.0", "result": 19, "id": 1})
+  """
+
+  require Logger
+
+  @doc """
+  Respond to a request for `method` with `params`.
+
+  You can return any serializable result (which will be ignored for notifications), or you can throw
```

Comment:
Throw? Really? That's so un-erlang.

Reply:
Thats just saying you can implement your handler this way. It doesn’t enforce it.

Reply:
On the other hand ... I don't think it's un-erlang like, lots of OTP throws exceptions for the simple purpose of control flow.

Reply:
really? Do you have examples? I just haven't run into that before.

Reply:
https://github.com/erlang/otp/blob/master/lib/ssl/src/ssl.erl#L1331

Reply:
Also an interesting comment (it's said it's quoting Virding but who knows), explaining diff between throws and errors: https://stackoverflow.com/a/13618490/10516731

Reply:
cool, thanks for the example! Still would prefer to avoid this practice, but not enough to say we should change the way this library has been conceived. Carry on. 👍